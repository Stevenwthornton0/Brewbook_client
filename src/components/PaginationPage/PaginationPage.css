.paginationContainer {
    margin-top: 30px;
}

.pagination {
    display: flex;
    flex-direction: row;
    width: fit-content;
    border: 1px solid black;
    border-radius: 4px;
}

.pagination > button {
    border: none;
    padding: 4px 10px;
    border-left: 1px solid black;
    font-size: 12px;
}

.pagination > .left {
    border-left: none;
    border-radius: 4px 0 0 4px;
}

.pagination > .right {
    border-radius: 0 4px 4px 0;
}

.disabled:hover {
    color: #bdbdbd;
    background-color: white;
    cursor: default;
    box-shadow: none;
}