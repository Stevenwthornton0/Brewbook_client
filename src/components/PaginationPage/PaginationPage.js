import React, { Component } from 'react';
import BreweryListContext from '../../contexts/BreweryListContext';
import Pagination from './Pagination/Pagination';
import './PaginationPage.css'

class PaginationPage extends Component {
    static contextType = BreweryListContext;

    constructor(props) {
        super(props)
        this.state = {
            renderPagination: null,
            pages: []
        }  
    }

    static defaultProps = {
        handlePagination: () => {},
        handlePrev: () => {},
        handleNext: () => {},
        currentPage: 1
    }

    componentDidMount() {
        this.setState({
            renderPagination: true
        })
    }


    createPagination = () => {
        const { maxPage, currentPage } = this.props;
        const paginationPage = () => {
            const { pages } = this.context;
            return pages.map(page => 
                <Pagination
                    currentPage={this.props.currentPage} 
                    key={page}
                    page={page}
                    handlePagination={this.props.handlePagination}
                />
            )
        }
        
        if (maxPage === 1 || maxPage === 0) {
            return null
        }
        
        if (currentPage === 1) {
            return (
                <div className='pagination'>
                    <button className='left disabled' disabled={true}>prev</button>
                    {paginationPage()}
                    <button className='right' onClick={() => {this.props.handleNext()}}>next</button>
                </div>
            )

        } else if (currentPage === maxPage) {
            return (
                <div className='pagination'>
                    <button className='left' onClick={() => {this.props.handlePrev()}}>prev</button>
                    {paginationPage()}
                    <button className='right disabled' disabled={true}>next</button>
                </div>
            )
        }

        return (
            <div className='pagination'>
                <button className='left' onClick={() => {this.props.handlePrev()}}>prev</button>
                {paginationPage()}
                <button className='right' onClick={() => {this.props.handleNext()}}>next</button>
            </div>
        )

    }
    
    render() {
        return (
            <div className='paginationContainer'>
                {this.state.renderPagination ? this.createPagination() : null}
            </div>
        )
    }
}

export default PaginationPage;

// creates a pagination based on the amount of pages and responses to a search