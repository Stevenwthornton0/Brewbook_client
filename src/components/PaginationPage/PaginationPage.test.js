import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import PaginationPage from './PaginationPage';

Enzyme.configure({ adapter: new Adapter() })

describe(`PaginationPage smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = mount(<PaginationPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})