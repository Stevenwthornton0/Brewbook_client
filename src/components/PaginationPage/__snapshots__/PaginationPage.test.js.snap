// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationPage smoke test renders without crashing 1`] = `
<PaginationPage
  currentPage={1}
  handleNext={[Function]}
  handlePagination={[Function]}
  handlePrev={[Function]}
>
  <div
    className="paginationContainer"
  >
    <div
      className="pagination"
    >
      <button
        className="left disabled"
        disabled={true}
      >
        prev
      </button>
      <button
        className="right"
        onClick={[Function]}
      >
        next
      </button>
    </div>
  </div>
</PaginationPage>
`;
